core/chapters/c12_dictionaries.py

```diff
@@ -0,0 +1,510 @@
+        """
+That doesn't work because the position of items in a dictionary usually doesn't matter.
+
+You don't usually care what's the 2nd or 5th or 100th word of the dictionary,
+you just want to find a specific word like 'apple'. So try that instead:
+
+__program_indented__
+        """
+
+        auto_translate_program = False
+
+        program = "french['apple']"
+
+        def check(self):
+            return "pomme" in self.result
+
+    class dict_access3(VerbatimStep):
+        """
+That's better!
+
+Now run a similar line in the shell to look up the translation for 'box'.
+
+__program_indented__
+        """
+        ##TODO: Hide the program from the user but still let them proceed through the excercise
+        auto_translate_program = False
+
+        program = "french['box']"
+
+        def check(self):
+            return "boite" in self.result
+
+    class dict_access4(VerbatimStep):
+        """
+And now you know both Python and French!
+
+Now let's translate from French to English:
+
+__program_indented__
+        """
+
+        auto_translate_program = False
+
+        program = "french['pomme']"
+
+        def check(self):
+            return "KeyError" in self.result
+
+    final_text = """
+Sorry, you can't do that either. You can only look up a key to get its value, not the other way around.
+
+The dictionary `french` only has 2 keys: `'apple'` and `'box'`. `'pomme'` is a value, not a key.
+
+We'll soon learn why you can't just look up values directly, and what you can do about it.
+
+
+Note that both `french[0]` and `french['pomme']` raised the same type of error: a `KeyError`.
+
+This error means that the provided key (`0` or `'pomme'` in this case) wasn't found in the dictionary.
+
+It's not that `french[0]` isn't *allowed*, it's just that it means the same thing as always:
+find the value associated with the key `0`. In this case it finds that no such key exists.
+
+But `0` *could* be a key, because many types of keys are allowed, including strings and numbers.
+"""
+
+
+class UsingDictionaries(Page):
+    class shopping_cart1(VerbatimStep):
+        """
+Let's see dictionaries in a real life problem. Imagine you're building an online shopping website.
+
+You keep the prices of all your items in a dictionary:
+
+__program_indented__
+        """
+
+        auto_translate_program = False
+
+        program = "prices = {'apple': 2, 'box': 5, 'cat': 100, 'dog': 100}"
+
+
+    class shopping_cart2(VerbatimStep):
+        """
+Here you can see one reason why looking up values in a dictionary could be a problem.
+
+What would `prices[100]` be? `'dog'`? `'cat'`? `['dog', 'cat']`?
+
+The same value can be repeated any number of times in a dictionary.
+
+On the other hand, keys have to be unique. Imagine if your prices started like this:
+
+__program_indented__
+        """
+
+        auto_translate_program = False
+
+        program = "prices = {'apple': 2, 'apple': 3}"
+
+    class shopping_cart3(ExerciseStep):
```

Comment:
These first three steps are all meant to be one step. Students aren't meant to run just `prices = {'apple': 2, 'apple': 3}`, nothing would happen.